conda/core/link.py

```diff
@@ -1003,28 +1001,21 @@ def run_script(prefix, prec, action='post-link', env_prefix=None, activate=False
             log.info("failed to run %s for %s due to COMSPEC KeyError", action, prec.dist_str())
             return False
         if activate:
-            conda_bat = env.get("CONDA_BAT", abspath(join(context.root_prefix, 'bin', 'conda')))
-            with tempfile.NamedTemporaryFile(
-                    mode='w', prefix=tmp_prefix, suffix='.bat', delete=False) as fh:
-                fh.write('@CALL \"{0}\" activate \"{1}\"\n'.format(conda_bat, prefix))
-                fh.write('echo "PATH: %PATH%\n')
-                fh.write('@CALL \"{0}\"\n'.format(path))
-                script_caller = fh.name
-            command_args = [comspec, '/d', '/c', script_caller]
+            command = '@CALL \"{0}\"\n'.format(path)
+            script_caller, command_args = wrap_subprocess_call(
+                on_win, context.root_prefix, prefix, command
+            )
         else:
             command_args = [comspec, '/d', '/c', path]
-
     else:
         shell_path = 'sh' if 'bsd' in sys.platform else 'bash'
         if activate:
-            conda_exe = env.get("CONDA_EXE", abspath(join(context.root_prefix, 'bin', 'conda')))
-            with tempfile.NamedTemporaryFile(mode='w', prefix=tmp_prefix, delete=False) as fh:
-                fh.write("eval \"$(\"{0}\" \"shell.posix\" \"hook\")\"\n".format(conda_exe)),
-                fh.write("conda activate \"{0}\"\n".format(prefix)),
-                fh.write("source \"{}\"\n".format(path))
-                script_caller = fh.name
-            command_args = [shell_path, "-x", script_caller]
+            command = ". \"{}\"\n".format(path)
```

Comment:
Calling path directly and not souring is correct here.  Thanks you for correcting this mistake from #8229.